### DISCUSSION TITLE
Error: page resources for /404/index.html not found

### DISCUSSION THREAD

**Discussion by konstantinschuette** [gatsbyjs/gatsby#36317] [Help] [answered]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

I found out that when I run lighthouse-ci and have the auto page (html file) detection turned on, two .html files for the 404 site will be tested.

1. `404/index.html`
2. `404.html`

The second one works fine, but the first one produces this error when visiting the site:
![image](https://user-images.githubusercontent.com/18708105/182676113-5a17f3e5-e695-4e3b-91b8-1148060982c4.png)

### Reproduction Link

https://githubbox.com/gatsbyjs/gatsby-starter-minimal

### Steps to Reproduce

1. Download `gatsby-starter-minimal` or just run `gatsby new`
2. `gatsby build`
3. `gatsby serve`
4. Open up your browser console
5. Visit `http://localhost:9000/404/index.html`

### Expected Result

When visiting `404/index.html` the error should not appear

### Actual Result

The error appears

### Environment

```shell
System:
    OS: Windows 10 10.0.19044
    CPU: (12) x64 Intel(R) Core(TM) i7-10850H CPU @ 2.70GHz
  Binaries:
    Node: 16.16.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 8.11.0 - C:\Program Files\nodejs\npm.CMD
  Languages:
    Python: 3.10.4
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (103.0.1264.77)
  npmPackages:
    gatsby: ^4.20.0 => 4.20.0
  npmGlobalPackages:
    gatsby-cli: 4.19.0
```

**Comment by LekoArts** [comment, marked as answer]
From https://github.com/GoogleChrome/lighthouse-ci/blob/main/docs/configuration.md#collect you can use the option `--autodiscoverUrlBlocklist`

> A URL to not include when autodiscovering urls from staticDistDir. Use this flag multiple times to filter multiple URLs.

Gatsby only creates these two files because sometimes hosts want a 404.html or a 404/index.html. During `gatsby serve` the 404.html is used, the other file is not stored in the pages store (as only ever one of those files will be used)

**Reply by konstantinschuette** [reply]
Okay thank you. If it's by design you can close the issue

**Reply by konstantinschuette** [reply]
https://www.gatsbyjs.com/docs/how-to/adding-common-features/add-404-page/

Wouldn't that be a good idear to add that specification to the docs?
